br/cmd/br/cmd.go

```diff
@@ -107,6 +110,23 @@ func AddFlags(cmd *cobra.Command) {
 	_ = cmd.PersistentFlags().MarkHidden(FlagRedactLog)
 }
 
+func calculateMemoryLimit(memleft uint64) uint64 {
```

Comment:
Do we follow the logic of tidb? if yes, can we reuse the their code?

Reply:
the tidb go memory limit tuner would reset the memory limit by TiDB node's `tidb_server_memory_limit`, which may large than BR node's memory.
For BRIE via SQL, BR task runs on a TiDB node, so it can use the memory limit tuner. 
For br clp, BR has forbidden the memory limit tuner. Besides, for br clp, the frequent GC will not affect the work of BR too much. Therefore, for simplicity, we can just fix the memory limit at the beginning.